backend/app/mcp_server/tools/knowledge.py

```
@@ -0,0 +1,770 @@
+    from app.services.context import context_service
+    from app.services.knowledge import KnowledgeService
+
+    attachment_id: int | None = None
+    try:
+        normalized_ext = _normalize_file_extension(file_extension)
+        filename = _build_filename(name, normalized_ext)
+
+        # Decode base64 content
+        try:
+            file_content = base64.b64decode(file_base64)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)